src/util.ts

```diff
@@ -1,10 +1,18 @@
-import { TextEditor } from "vscode";
+import { TextEditor, WorkspaceConfiguration } from "vscode";
 import {
   ExecuteCommandRequest,
   TextDocumentPositionParams,
 } from "vscode-languageclient";
 import { LanguageClient } from "vscode-languageclient/node";
 
+declare const sym: unique symbol;
+/**
+ * Creates a newtype without any runtime overhead. It's important for ID to be both unique and descriptive.
+ */
+export type newtype<A, ID extends string> = A & {
+  readonly [sym]: ID;
+};
+
```

Comment:
I know it's a little bit magical to the person who is not familiar with the typescript type system & magic. However, these types created by `newtype` produce compile errors and have meaningful names instead of just `string`.

Reply:
It seems like it's too much for poor Tomek :D

Reply:
I'm quite familiar with this encoding of newtypes, but I think the unique symbol should be the value, otherwise it's quite useless, the definition above is equivalent to:

```ts
export type newtype<A, ID extends string> = A & {
  readonly theType: ID;
};
```

This way the uniqueness is only guaranteed by the string passed when defining the type, which can work ok, but you are not using the unique symbol.

An alternative encoding can just be:

```ts
export type Newtype<A, O> = A & O

type UserId = Newtype<number, { readonly UserId: unique symbol }>
type ThingId = Newtype<number, { readonly ThingId: unique symbol}>
```

Full example here: 

https://www.typescriptlang.org/play?ssl=1&ssc=13&pln=1&pc=20#code/KYDwDg9gTgLgBDAnmYcBywDuSUB4CCANHAPIB8cAvHPnAGSkBQjOqAqgM7BQCSAJlXRZWuAHYBXALYAjbsQDecKMACGfCKIA2iOJ278AXHHGiAlgEdxqDohkRNcAL5kWyVABUAFqdEBzfoIY2G5iUrJQCkqq6lo6Xj7+fEYmFlZwNnaazsx8wADGmirKcABmJnkwphqlABTiXLxJug38AJRGAG4QpnwA3Iy5BUWoeRoc8PX6TXqN-YOFxaOi4wjefoZw8et9zCU1MGuJrYx7k42tQA

(this is loosely based on https://github.com/gcanti/newtype-ts/blob/master/src/index.ts)

Also (suuuper tiny detail) this is not strictly a newtype, since a `TargetName` can be passed where a `string` is expected. The encoding in newtype-ts is a bit stricter in this sense, that's why I've relaxed it a bit

Reply:
>I'm quite familiar with this encoding of newtypes, but I think the unique symbol should be the value, otherwise it's quite useless, the definition above is equivalent to:

Actually, I used that in this way intentionally. The symbol is declared as a unique one and is used as a key for unnamed property. Moreover, the symbol is not exported so only a newtype can use it. Thanks to that you cannot access this property or even fake it. What do I mean by "fake it"? Let's consider the following example:
```typescript
type newtype<A, ID extends string> = A & {
  readonly theType: ID;
};
type SuiteName = newtype<string, "suiteName">;
type Type2 = string & {
  theType: "suiteName";
};

const x: Type2 = "aezakmi" as Type2;
const y: Type1 = x; // works
```
I know that this is an almost never gonna happen scenario, but I try to be careful with structural typing.

>Also (suuuper tiny detail) this is not strictly a newtype, since a TargetName can be passed where a string is expected. The encoding in newtype-ts is a bit stricter in this sense, that's why I've relaxed it a bit

Yeah, you're right. But this is more than fine in this case since I wanted to use "newtype" to distinguish between string types and for us, the name `newtype` shouldn't be misleading.

Reply:
Thanks for the explanation, it make sense then! One minor thing is that not using a unique symbol in the value too means that you can accidentally do:

type Type1 = newtype<string, "type1">;
type Type2 = newtype<string, "type1">; // oops

and those two will be silently interchangeable.

It's probably ok if we don't have many cases, it's a mistake that can be caught in code review